Jint/Engine.cs

```diff
@@ -369,6 +400,131 @@ public Engine Execute(Script script)
             return this;
         }
 
+        /// <summary>
+        /// EXPERIMENTAL! Subject to change.
+        /// 
+        /// Executes a script and then tries to process all schedules events.
+        /// It triggers onFinished callback when there are no scheduled events in the EventLoop
+        /// and when there are no user registered promises left unresolved.
+        /// </summary>
+        /// <param name="script"></param>
+        /// <param name="onFinished">called when EventLoop had no more events neither pending user promises</param>
+        /// <returns></returns>
+        public Engine ExecuteWithEventLoop(Script script, Action onFinished)
```

Comment:
What is a use case for `onFinished`?

Reply:
`onFinished` is a way to determine if everything is settled, but it can probably be tracked in user space, because the only reason why it can be not settled are user created promises

Reply:
killed it